scripts/Dockerfile.e2e

```diff
@@ -0,0 +1,76 @@
+# syntax=docker/dockerfile:1.7-labs
+# (use non-stable syntax for convenient --parents option in COPY command)
+
+# Download clang
+FROM scratch AS clang
+ADD https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.4/clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04.tar.xz /clang.tar.xz
+
+FROM ubuntu:latest
+
+#####################
+# DEPENDENCIES
+#####################
+# Install misc build dependencies
+# TODO: only install required hard requirements, see best practices
+# https://docs.docker.com/build/building/best-practices/#apt-get
+RUN apt-get update && \
+    apt-get install -y -qq \
+        apt-transport-https \
+        bash \
+        binaryen \
+        bison \
+        build-essential \
+        curl \
+        g++ \
+        g++-i686-linux-gnu \
+        gawk \
+        gcc \
+        gcc-i686-linux-gnu \
+        git \
+        gnupg \
+        golang \
+        libssl-dev \
+        libxml2 \
+        openssl \
+        python3 \
+        sudo \
+        unzip \
+        vim \
+        wget \
+        zip
```

Comment:
We could save a little bit of room here if we clean up after the updates with `rm -rf /var/lib/apt/lists/*`